bin/pycbc_make_sky_grid

```diff
@@ -62,92 +71,62 @@ def make_multi_det_grid(args):
     detectors = [Detector(d) for d in detectors]
     detector_pairs = list(itertools.combinations(detectors, 2))
 
-    # Calculate the time delay for each detector pair
-    tds = [
-        detector_pairs[i][0].time_delay_from_detector(
-            detector_pairs[i][1], args.ra, args.dec, args.trigger_time
-        )
-        for i in range(len(detector_pairs))
-    ]
-
-    # Calculate the light travel time between the detector pairs
-    light_travel_times = [
-        detector_pairs[i][0].light_travel_time_to_detector(detector_pairs[i][1])
-        for i in range(len(detector_pairs))
-    ]
-
-    # Calculate the required angular spacing between the sky points
-    ang_spacings = [
-        (2 * args.timing_uncertainty)
-        / np.sqrt(light_travel_times[i] ** 2 - tds[i] ** 2)
-        for i in range(len(detector_pairs))
-    ]
-    angular_spacing = min(ang_spacings)
-
-    sky_points = np.zeros((1, 2))
-
-    number_of_rings = int(args.sky_error / angular_spacing)
-
-    # Generate the sky grid centered at the North pole
-    for i in range(number_of_rings + 1):
-        if i == 0:
-            sky_points[0][0] = 0
-            sky_points[0][1] = np.pi / 2
-        else:
-            number_of_points = int(2 * np.pi * i)
-            for j in range(number_of_points):
-                sky_points = np.row_stack(
-                    (
-                        sky_points,
-                        np.array([j / i, np.pi / 2 - i * angular_spacing]),
-                    )
+    input_dist = eval(args.input_dist)
+    sky_dist = eval(args.input_dist)
+    sky_dist = sky_dist.to_uniform_patch(args.coverage) 
+    grid = np.empty((0,2))
+
+    while True:
+        prev_size = grid.shape[0]
+        sky_dist_samples = sky_dist.rvs(size=10000)
+        sky_ra = sky_dist_samples['ra']
+        sky_dec = sky_dist_samples['dec']
+        for prop_ra, prop_dec in zip(sky_ra, sky_dec):
+            # Calculate the time delay for each detector pair
+            tds = [
+                detector_pairs[i][0].time_delay_from_detector(
+                    detector_pairs[i][1], prop_ra, prop_dec, args.trigger_time
                 )
-
-    # Convert spherical coordinates to cartesian coordinates
-    cart = spher_to_cart(sky_points)
-
-    grb = np.zeros((1, 2))
-    grb[0] = args.ra, args.dec
-    grb_cart = spher_to_cart(grb)
-
-    north_pole = [0, 0, 1]
-
-    ort = np.cross(grb_cart, north_pole)
-    norm = np.linalg.norm(ort)
-    ort /= norm
-    n = -np.arccos(np.dot(grb_cart, north_pole))
-    u = ort * n
-
-    # Rotate the sky grid to the center of the external trigger error box
-    r = R.from_rotvec(u)
-    rota = r.apply(cart)
-
-    # Convert cartesian coordinates back to spherical coordinates
-    spher = cart_to_spher(rota)
-
-    sky_grid = SkyGrid(
-        spher[:, 0], spher[:, 1], args.instruments, args.trigger_time
-    )
-    return sky_grid
+                for i in range(len(detector_pairs))
```

Comment:
You can simplify this code if you iterate directly over `detector_pairs` instead of over its index:
```
a.time_delay_from_detector(b, …) for a, b in detector_pairs
```